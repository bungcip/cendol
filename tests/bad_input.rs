use cendol::test_utils::compile_and_get_error;
use insta::assert_yaml_snapshot;

#[test]
fn test_unexpected_token() {
    let err = compile_and_get_error("int main() { int a = 1 +; }", "unexpected_token.c");
    assert_yaml_snapshot!("unexpected_token", err.unwrap_err());
}

#[test]
fn test_invalid_variable_initialization() {
    let err = compile_and_get_error(
        r#"
int main() {
    int x = "hello";
    return 0;
}
"#,
        "invalid_variable_initialization.c",
    );
    assert_yaml_snapshot!("invalid_variable_initialization", err.unwrap_err());
}
