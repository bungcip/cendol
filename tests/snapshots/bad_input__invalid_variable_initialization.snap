---
source: tests/bad_input.rs
expression: err.unwrap_err()
---
msg: "Invalid initializer: cannot initialize variable of type `int` with value of type `char*`"
path: invalid_variable_initialization.c
span:
  file: 0
  start:
    file: 0
    line: 3
    column: 13
  end:
    file: 0
    line: 3
    column: 19
verbose: false
